bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java

```diff
@@ -91,38 +92,43 @@ public void onCompleted() {
           return streamRows(request);
         }
       };
-  private final RetryableRpc<SampleRowKeysRequest, List<SampleRowKeysResponse>> sampleRowKeysAsync;
-  private final RetryableRpc<ReadRowsRequest, List<Row>> readRowsAsync;
+  private RetryableRpc<SampleRowKeysRequest, List<SampleRowKeysResponse>> sampleRowKeysAsync;
+  private RetryableRpc<ReadRowsRequest, List<Row>> readRowsAsync;
+
+  @VisibleForTesting
+  private ClientCallService clientCallService;
 
   public BigtableDataGrpcClient(
       Channel channel,
       ExecutorService executorService,
-      RetryOptions retryOptions) {
+      RetryOptions retryOptions,
+      ClientCallService clientCallService) {
```

Comment:
Do we want to expose ClientCallService?

how about public constructor without ClientCallService and package default for testing with ClientCallService?